queue/message_prioritizer_test.go

```diff
@@ -0,0 +1,252 @@
+package queue
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/aquasecurity/table"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/ssv"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	mockStateRunning = &State{
+		HasRunningInstance: true,
+		Height:             100,
+		Slot:               64,
+		Quorum:             4,
+	}
+)
+
+func TestMessagePrioritizerSlice(t *testing.T) {
+	tests := []struct {
+		name     string
+		state    *State
+		messages []mockMessage
+	}{
+		{
+			name:  "All-in-one",
```

Comment:
Would be better to check specific cases in addition to all-in-one

Reply:
agreed. need to add tests based on exist spec-tests

Reply:
Any ideas for new tests? Or should we reconsider this after implementation in SSV?
@amir-blox @nivBlox